Tests running.

namangoel1 opened the pull request:
The test wait for a mock email to go out, and check if there's any error or was it actually sent.

namanyayg committed:
Tests running.

The test wait for a mock email to go out, and check if there's
any error or was it actually sent.

roonyh commented on tests/testRootPost.js:
```diff
@@ -3,11 +3,30 @@ var routes = require('../routes/index');
+    routes.rootPost(testRequest, testResponse);
+
+
+    setTimeout(function() {
```
You can avoid this by moving the test code into the render function. Render does not really need to be a spy. It will always be called after the mail is sent. So we can place the code that tests `flash.args` there.

roonyh commented on tests/testRootPost.js:
```diff
@@ -3,11 +3,30 @@ var routes = require('../routes/index');
+
+    setTimeout(function() {
+      if ( testResponse.render.calledOnce ) {
+        console.log( '    ', testRequest.flash.args[0][1] )
```
instead of console.logging we can tests those using `assert`

roonyh replied:
http://nodejs.org/api/assert.html

namangoel1 replied:
I wanted the user to see the message (it could be an error also, after all). Is console.log not appropriate here even then?

namanyayg committed:
Mock stub test works

Mock test uses sinon.stub() and tests if the bodies are the same.
Renaming the old test.

namanyayg committed:
Fixing testEmailRootPost to be cleaner and optimized

namanyayg committed:
Merging testEmailRootPost and testRootPost

testRootPost.js is now a more comprehensive test that checks if
'render' is called on the rootPost function properly

namanyayg committed:
Using assert() instead of if()
